src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length; i++) {
+            if (i < figures.length / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+                System.out.println("Random figure: " + figures[i]);
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+                System.out.println("Default figure: " + figures[i]);
+            }

```

Comment:
Sorry, maybe I wasn't clear enough. Previous version with .draw() method usage was correct in general.
What were required to do is just extract it from conditions to use one time. Like:
if (condition) {
    figures[I] = ...
} else {
    figures[I] = ...
}
figures[I].draw()